MaxText/maxtext_utils.py

```diff
@@ -82,3 +94,98 @@ def get_train_input_output_trees(func, input_args, input_kwargs):
   in_tree, out_tree = get_train_input_output_trees(partial_train, shaped_input_args, shaped_input_kwargs)
   p_train_step = deserialize_and_load(serialized_compiled, in_tree, out_tree)
   return p_train_step
+
+def adam_pax(
```

Comment:
is this not the same as Optax Adam? If possible, we'd like to avoid owning this code. If impossible, please put this in optimizers.py

Reply:
Verified they are not equivalent with subtle differences in orders of momentum updates. Got convergence issue in using optax adam. and optax adam doesn't support weight decay as well. Will put it in a new optimizers.py.

Reply:
Done.